src/backend/utils/adt/ruleutils.c

```diff
@@ -11117,7 +11117,7 @@ get_partition_recursive(PartitionNode *pn, deparse_context *head,
 					{
 						first_every_rule = NULL;
 
-						if (!strlen(rule->parname))
+						if (!rule->parname || !strlen(rule->parname))
```

Comment:
How about to do same thing on similar [line](https://github.com/arenadata/gpdb/pull/698/files#diff-6c12827097e95912476267d73f3f56e8250b13aad37ab4286c3c7fdab7f30ec7R11013)?

Reply:
done